src/components/token/TokenBalances.js

```diff
@@ -83,14 +93,21 @@ class TokenBalances extends React.Component {
     await helpers.setStateAsync(this, {
       sortBy: get(queryParams, 'sortBy', this.state.sortBy),
       order: get(queryParams, 'order', this.state.order),
+      tokenId: get(queryParams, 'token', this.state.tokenId),
       loading: true,
+      queryParamsParsed: true,
```

Comment:
I think it would be possible to move the line `const queryParams = this.pagination.obtainQueryParams();` to the `constructor`, and set the initial state there already with the right parameters.

This would probably avoid needing to create this `queryParamsParsed` variable.

Reply:
Yeah, you are right. Did this refactor here: https://github.com/HathorNetwork/hathor-explorer/pull/201/commits/ba4efc5acf5cb611f38a7c596b2d5b364ca3e26b